apps/designer/app/designer/features/style-panel/style-section.tsx

```diff
@@ -0,0 +1,780 @@
+  const Control = controls[styleConfig.control];
+  const { property } = styleConfig;
+  const key = category + "-" + property;
+  if (!Control) return null;
+  return (
+    <Box
+      key={key}
+      data-type={styleConfig.control.toLowerCase()}
+      data-property={property}
+      css={{ gridArea: property }}
+    >
+      <Control
+        currentStyle={currentStyle}
+        inheritedStyle={inheritedStyle}
+        setProperty={setProperty}
+        styleConfig={styleConfig}
+        category={category}
+      />
+    </Box>
+  );
+};
+
+type RenderCategoryProps = {
+  setProperty: SetProperty;
+  createBatchUpdate: CreateBatchUpdate;
+  currentStyle: Style;
+  category: Category;
+  styleConfigsByCategory: Array<JSX.Element | null>;
+  moreStyleConfigsByCategory: Array<JSX.Element | null>;
+};
+// Categories should render themselves because most Categories will not be flat un-ordered lists with
+// the new designs, refactor ColorControl, SpacingControl if needed.
+export const renderCategory = ({
+  setProperty,
+  createBatchUpdate,
+  currentStyle,
+  category,
+  styleConfigsByCategory,
+  moreStyleConfigsByCategory,
+}: RenderCategoryProps) => {
+  switch (category) {
+    case "layout": {
+      const css = {
+        alignItems: "center",
+        gap: "8px",
+        "& > [data-type=iconbuttonwithmenu]": {
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100%",
+        },
+      };
+      switch (currentStyle.display?.value) {
+        case "inline-flex":
+        case "flex": {
+          return (
+            <>
+              <Grid
+                css={{
+                  ...css,
+                  gridTemplateColumns: "repeat(12, 1fr)",
+                  gridTemplateRows: "auto 0px auto auto 0px auto",
+                  gridTemplateAreas: `
+                  "${"display ".repeat(12)}"
+                  "${"grid ".repeat(5)} . . . . . . ."
+                  "grid grid grid grid grid flexDirection flexDirection flexWrap flexWrap justifyItems justifyItems ."
+                  "grid grid grid grid grid alignItems alignItems justifyContent justifyContent alignContent alignContent ."
+                  "${"grid ".repeat(5)} . . . . . . ."
+                  "${"columnGap ".repeat(5)} lock lock ${"rowGap ".repeat(5)}"
+                `,
+                  // @todo justifyItems icons are using justifycontent icons atm
+                  "& > [data-property=justifyItems]": {
+                    display: "none",
+                  },
+                }}
+              >
+                {styleConfigsByCategory}
+                <LockControl
+                  data-property="lock"
+                  css={{ gridArea: "lock" }}
+                  currentStyle={currentStyle}
+                  setProperty={setProperty}
+                />
+                <GridControl
+                  data-property="grid"
+                  css={{ gridArea: "grid" }}
```

Comment:
same here

Reply:
in fact you could get rid of both, CSS and data-property by passing one name and rendering it same way